core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -86,22 +86,23 @@ oppia.factory('responsesService', [
             .can_have_solution && stateSolutionService.savedMemento !== null) {
           if (stateSolutionService.savedMemento.correctAnswer !== null) {
             var currentStateName = editorContextService.getActiveStateName();
-            SolutionHelperService.verifySolution(
+            SolutionVerificationService.verifySolution(
               explorationContextService.getExplorationId(),
               explorationStatesService.getState(currentStateName),
               stateSolutionService.savedMemento.correctAnswer,
               function() {
                 explorationStatesService.updateSolutionValidity(
-                  currentStateName, false);
+                  currentStateName, true);
                 alertsService.addInfoMessage(
-                  'That solution does not lead to the next state!');
+                  'The solution is now valid!');
```

Comment:
I'm sending callback functions for success and error cases. If I need to handle more than just the error/success scenario there will need to be some branching (and confusion). I think the warning icon is good enough.

Or I could show just one success/error message which will make sense in either scenario. ie. "Solution specified is valid!" / "Solution specified is invalid!"